rating-service/src/main/java/edu/miu/ratingservice/service/impl/RatingServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package edu.miu.ratingservice.service.impl;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.miu.ratingservice.dto.request.RatingRequestDTO;
+import edu.miu.ratingservice.dto.response.RatingResponseDTO;
+import edu.miu.ratingservice.model.Rating;
+import edu.miu.ratingservice.repository.RatingRepository;
+import edu.miu.ratingservice.service.RatingService;
+import edu.miu.ratingservice.utils.RatingUtils;
+
+@Service
+@Transactional
+public class RatingServiceImpl implements RatingService {
+    private final RatingRepository ratingRepository;
+
+    public RatingServiceImpl(RatingRepository ratingRepository) {
+        this.ratingRepository = ratingRepository;
+    }
+
+    // @Override
+    // public List<RatingResponseDTO> getRatingsByMeal(Long id) {
+
+    // return null;
+    // }
+
+    // @Override
+    // public List<RatingResponseDTO> getRatingsByDay() {
+    // return null;
+    // }
+
+    // @Override
+    // public List<RatingResponseDTO> getRatingsByUser(Long id) {
+    // Rating rating = ratingRepository.findByUser(id).get();
+    // return (List<RatingResponseDTO>)
+    // RatingUtils.parseRatingToRatingReponseDTOObject(rating);
+    // }
+
+    @Override
+    public RatingResponseDTO getRatingById(Long id) {
+        Rating rating = ratingRepository.findById(id).get();
+        return RatingUtils.parseRatingToRatingReponseDTOObject(rating);
+    }
+
+    @Override
+    public String addRating(RatingRequestDTO ratingRequestDTO) {
+        Rating rating = RatingUtils.parseRatingRequestDTOToRating(ratingRequestDTO);
+        rating = ratingRepository.save(rating);
+        return "Rating Added";
```

Comment:
Returning a String is not useful for us. Should return the newly added Rating instead, ideally the responseDTO.